File: src/test/java/com/bbcnews/automation/scripts/BBCNewsHindiSmokeTest.java
package com.bbcnews.automation.scripts;

import com.aventstack.extentreports.ExtentTest;
import com.bbcnews.automation.commonfunctions.CommonFunctions;
import com.bbcnews.automation.pageobjects.*;
import com.bbcnews.automation.testutils.AppiumStart;
import com.bbcnews.automation.testutils.ExtenttestReport;
import io.appium.java_client.MobileElement;
import io.appium.java_client.android.AndroidDriver;
import io.appium.java_client.pagefactory.AppiumFieldDecorator;
import io.appium.java_client.remote.MobileCapabilityType;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import io.qameta.allure.Story;
import org.openqa.selenium.ScreenOrientation;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.support.PageFactory;
import org.testng.Assert;
import org.testng.ITestResult;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import java.io.File;
import java.io.IOException;
import java.net.URL;
import java.text.SimpleDateFormat;
import java.util.Date;

public class BBCNewsHindiSmokeTest {


        private AndroidDriver<MobileElement> androidDriver;
        private DesiredCapabilities capabilities;
        private String Deviceos_Name;
        private String Device_id;
        private String Device_Name;
        private String App_Path;
        private String Appium_Port;
        private File file;
        ExtentTest test;

        private HomePageObject homePageObject;
        private MyNewsPageObject myNewsPageObject;
        private BasePageObject basePageObject;
        private VidoePageObject vidoePageObject;
        private PopularPageObject popularPageObject;

        ExtenttestReport extenttestReport = new ExtenttestReport();
        AppiumStart appiumStart = new AppiumStart();

        private static Date curDate = new Date();
        private static SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
        private static String DateToStr = format.format(curDate);
        public static String workingDirectory = System.getProperty("user.dir");
        private static String screenshotpath = workingDirectory + "/Screenshots/";

        @BeforeTest
        public void RunTest() throws Exception
    {

        try {

            readDeviceDetailsCommandPrompt();
            setUP();
            LaunchBBCNews();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

        public void readDeviceDetailsCommandPrompt()
    {

        try {
            Deviceos_Name = System.getProperty("DeviceOS");
            Device_id = System.getProperty("DeviceID");
            Device_Name = System.getProperty("DeviceName");
            App_Path = System.getProperty("AppPath");
            Appium_Port = System.getProperty("AppiumPort");
            System.out.println("Passed The Device OS is " + Deviceos_Name);
            System.out.println("Passed The Device ID is " + Device_id);
            System.out.println("Passed The Device Name is " + Device_Name);
            System.out.println("Passed The Appium port is " + Appium_Port);
            System.out.println("Passed The Application path  is " + App_Path);
        } catch (Exception e) {
            e.printStackTrace();

        }

    }


        public void setUP()
    {
        try {
            //  appiumStart.startAppium(Integer.parseInt(Appium_Port));
            String appium_url = "http://127.0.0.1:" + Appium_Port + "/wd/hub";
            System.out.println("Appium Server Address : - " + appium_url);
            capabilities = new DesiredCapabilities();
            capabilities.setCapability(MobileCapabilityType.UDID, Device_id);
            capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "bbcnews");
            capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, "UiAutomator2");
            capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android");
            capabilities.setCapability("appiumversion", "1.8.1");
            capabilities.setCapability("app", App_Path); //"/Users/ramakh02/Desktop/tools/APK/BBCNews-5.5.0.35.apk");
            //it's not mandatory to pass OS version of the device
            // capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION,Deviceos_Name);
            capabilities.setCapability("appPackage", "uk.co.bbc.hindi.internal");
            capabilities.setCapability("appActivity", "bbc.mobile.news.v3.app.TopLevelActivity");
            // capabilities.setCapability(MobileCapabilityType.UNEXPECTED_ALERT_BEHAVIOUR, true);
            // capabilities.setCapability("autoAcceptAlerts", true);
            capabilities.setCapability("--session-override", true);
            androidDriver = new AndroidDriver<MobileElement>(new URL(appium_url), capabilities);
             } catch (Exception e)
                {
                }
    }

        public void LaunchBBCNews()
    {
        try {
            homePageObject = new HomePageObject();
            PageFactory.initElements(new AppiumFieldDecorator(androidDriver), homePageObject);

            myNewsPageObject = new MyNewsPageObject();
            PageFactory.initElements(new AppiumFieldDecorator(androidDriver), myNewsPageObject);

            basePageObject = new BasePageObject();
            PageFactory.initElements(new AppiumFieldDecorator(androidDriver), basePageObject);

            vidoePageObject = new VidoePageObject();
            PageFactory.initElements(new AppiumFieldDecorator(androidDriver), vidoePageObject);

            popularPageObject = new PopularPageObject();
            PageFactory.initElements(new AppiumFieldDecorator(androidDriver), popularPageObject);

            CommonFunctions.emptyFolder(screenshotpath);

            extenttestReport.createrReportHive("SmokeTest", Deviceos_Name, Device_Name, Device_id);

            androidDriver.context("NATIVE_APP");
            file = new File(screenshotpath);
            String screenshot = file.getAbsolutePath();
            System.out.println("The ScreenShot Path is " + screenshot);

            ScreenOrientation orientation = androidDriver.getOrientation();
            if (orientation.equals(ScreenOrientation.LANDSCAPE)) {
                androidDriver.rotate(ScreenOrientation.PORTRAIT);
            }


        } catch (NullPointerException e)
            {
                 e.printStackTrace();
            } catch (Exception e) {
            e.printStackTrace();
                    }
    }


    @Test(priority = 1, description = "launching the app ")
    @Story("Home")
    @Severity(SeverityLevel.CRITICAL)
    public void testOpenNewsApp() throws Exception {
        try {

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_okbutton,false);
            extenttestReport.tapButton(androidDriver,basePageObject.nothanksbutton,false);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test(priority = 2, description = "Check the links on the Home page after app launched")
    @Story("Home")
    @Severity(SeverityLevel.CRITICAL)
    public void testCheckHomePage(){
        try {
            extenttestReport.startTest("HomePage", "Checking the HomePage", "Smoke");
            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_homepage,false);
            Assert.assertTrue(basePageObject.bbchindi_homepage.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_homepage);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_india,false);
            Assert.assertTrue(basePageObject.bbchindi_india.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_india);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_international,false);
            Assert.assertTrue(basePageObject.bbchindi_international.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_international);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_entertainment,false);
            Assert.assertTrue(basePageObject.bbchindi_entertainment.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_entertainment);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_sports,false);
            Assert.assertTrue(basePageObject.bbchindi_sports.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_sports);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_radio,false);
            Assert.assertTrue(basePageObject.bbchindi_radio.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_radio);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_sciencetechnology,false);
            Assert.assertTrue(basePageObject.bbchindi_sciencetechnology.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_sciencetechnology);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_lookat,false);
            Assert.assertTrue(basePageObject.bbchindi_lookat.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_lookat);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_thephotos,false);
            Assert.assertTrue(basePageObject.bbchindi_thephotos.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_thephotos);

            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_social,false);
            Assert.assertTrue(basePageObject.bbchindi_social.isSelected());
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_social);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }


    @Test(priority = 3, description = "Checking the MoreOptions Menu")
    @Story("MoreOptions")
    @Severity(SeverityLevel.CRITICAL)
    public void testMenuItems() throws Exception {
        try {
            extenttestReport.startTest("MoreOptions", "Checking the MoreOptions Menu", "Smoke");
            extenttestReport.tapButton(androidDriver,basePageObject.bbc_moreoptions,false);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_help);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_Internalsettings);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_settings);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_pleasecontact);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_OtherBBCapplications);
            extenttestReport.navigateBack(androidDriver);


        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test(priority = 4, description = "Checking the More Settings Options Menu")
    @Story("MoreOptions")
    @Severity(SeverityLevel.CRITICAL)
    public void testMoreSettingsOptions() throws Exception {
        try {
            extenttestReport.startTest("MoreOptions", "Checking the More Settings Options Menu", "Smoke");
            extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_Moresettings,false);
           // extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_localnews);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.bbchindi_topics);



        } catch (Exception e) {
            e.printStackTrace();
        }
    }


    @Test(priority = 5, description = "Checking the More Settings Options Topics")
    @Story("MoreOptions-Topics")
    @Severity(SeverityLevel.CRITICAL)
    public void testMoreSettingsOptions_Topics() throws Exception {
        try {
            extenttestReport.startTest("MoreOptionsTopics", "Checking the More Settings Options Topics", "Smoke");
          //  extenttestReport.tapButton(androidDriver,basePageObject.bbchindi_topics_collapsegroup,false);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindihomepage);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindibharath);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindienrairnment);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindiinternatonal);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindilookat);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindiphotos);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindiscience);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindiphotos);
            extenttestReport.elementDisplayed(androidDriver,basePageObject.hindisocial);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @AfterMethod
    public void getResult(ITestResult result)
    {
        try {
            extenttestReport.getTestResult(androidDriver,result);
        }catch (IOException e)
        {}


    }
    @AfterTest
    public void tearDown()
    {
        extenttestReport.publishReport();
        androidDriver.closeApp();
        androidDriver.quit();

    }
}

File: src/test/java/com/bbcnews/automation/scripts/BBCNewsHindiSmokeTest.kt
package com.bbcnews.automation.scripts

import com.aventstack.extentreports.ExtentTest
import com.bbcnews.automation.commonfunctions.CommonFunctionKotlin
import com.bbcnews.automation.pageobjects.*
import com.bbcnews.automation.testutils.Testutility
import io.appium.java_client.MobileElement
import io.appium.java_client.android.AndroidDriver
import io.appium.java_client.pagefactory.AppiumFieldDecorator
import io.appium.java_client.remote.MobileCapabilityType
import io.qameta.allure.Severity
import io.qameta.allure.SeverityLevel
import io.qameta.allure.Story
import org.openqa.selenium.By
import org.openqa.selenium.ScreenOrientation
import org.openqa.selenium.remote.DesiredCapabilities
import org.openqa.selenium.support.PageFactory
import org.testng.Assert
import org.testng.Assert.assertTrue
import org.testng.ITestResult
import org.testng.annotations.AfterMethod
import org.testng.annotations.AfterTest
import org.testng.annotations.BeforeTest
import org.testng.annotations.Test

import java.io.File
import java.io.IOException
import java.net.URL
import java.text.SimpleDateFormat
import java.util.Date

class BBCNewsHindiSmokeTest {



    private var capabilities = DesiredCapabilities()

    private var deviceid: String? = null
    private var deviceName: String? = null
    private var AppPath: String? = null
    private var AppiumPort: String? = null
    private lateinit var file: File
    internal var test: ExtentTest? = null

    private lateinit var homePageObject: HomePageObject
    private lateinit var androidDriver: AndroidDriver<MobileElement>
    private lateinit var bbcNewsHindiPageObject: BBCNewsHindiPageObject

    private var commonFunctionKotlin = CommonFunctionKotlin()
    private var testutility = Testutility()


    @BeforeTest
    @Throws(Exception::class)
    fun runTest() {

        try {

            readDeviceDetailsCommandPrompt()
            setUP()
            commonFunctionKotlin.checkConnection(androidDriver)
            launchBBCNews()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

   private  fun readDeviceDetailsCommandPrompt() = try {

        deviceid = System.getProperty("DeviceID")
        deviceName = System.getProperty("DeviceName")
        AppPath = System.getProperty("AppPath")
        AppiumPort = System.getProperty("AppiumPort")
        System.out.println("Passed The Device ID is $deviceid")
        System.out.println("Passed The Device Name is $deviceName")
        System.out.println("Passed The Appium port is $AppiumPort")
        System.out.println("Passed The Application path  is $AppPath")
    } catch (e: Exception) {
        e.printStackTrace()

    }


   private fun setUP() {
        try {

            val appium_url = "http://127.0.0.1:$AppiumPort/wd/hub"
            System.out.println("Appium Server Address : - $appium_url")
            capabilities = DesiredCapabilities()
            capabilities.setCapability(MobileCapabilityType.UDID, deviceid)
            capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "bbcnews")
            capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, "UiAutomator2")
            capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "Android")
            capabilities.setCapability("appiumversion", "1.8.1")
            capabilities.setCapability("app", AppPath)
            capabilities.setCapability("appPackage", "uk.co.bbc.hindi.internal")
            capabilities.setCapability("appActivity", "bbc.mobile.news.v3.app.TopLevelActivity")
            capabilities.setCapability(MobileCapabilityType.UNEXPECTED_ALERT_BEHAVIOUR, true)
            capabilities.setCapability("autoAcceptAlerts", true)
            capabilities.setCapability("--session-override", true)
            androidDriver = AndroidDriver(URL(appium_url), capabilities)
        } catch (e: Exception) {
        }

    }

   private fun launchBBCNews() {
        try {
            homePageObject = HomePageObject()
            PageFactory.initElements(AppiumFieldDecorator(androidDriver), homePageObject)

            bbcNewsHindiPageObject = BBCNewsHindiPageObject()
            PageFactory.initElements(AppiumFieldDecorator(androidDriver), bbcNewsHindiPageObject)

            testutility.emptyFolder(screenshotpath)

            commonFunctionKotlin.createrReportHive("Regression", deviceName.toString(), deviceid.toString())

            androidDriver.context("NATIVE_APP")
            file = File(screenshotpath)
            val screenshot = file.absolutePath
            System.out.println("The ScreenShot Path is $screenshot")

            val orientation = androidDriver.orientation
            if (orientation == ScreenOrientation.LANDSCAPE) {
                androidDriver.rotate(ScreenOrientation.PORTRAIT)
            }


        } catch (e: NullPointerException) {
            e.printStackTrace()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }


    @Test(priority = 1, description = "launching the app ")
    @Story("Home")
    @Severity(SeverityLevel.CRITICAL)
    @Throws(Exception::class)
    fun testOpenNewsApp() {
        try {

            Assert.assertEquals("ओके", bbcNewsHindiPageObject.bbchindi_okbutton.text, "Text Matched")
            Assert.assertEquals("बीबीसी न्यूज़ आपको नोटिफ़िकेशंस भेजना चाहता है. आप कभी भी सेटिंग्स में जाकर बदलाव कर सकते हैं.", bbcNewsHindiPageObject.bbchindi_message.text)
            commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_okbutton, false)
            commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.nothanksbutton, false)

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

   @Test(priority = 2, description = "Check the links on the Home page after app launched")
   fun testCheckHomePage() = try {
       commonFunctionKotlin.startTest("HomePage", "Checking the HomePage", "Smoke");
       commonFunctionKotlin.tapButton(androidDriver,bbcNewsHindiPageObject.bbchindi_homepage,false)
       assertTrue(bbcNewsHindiPageObject.bbchindi_homepage.isSelected)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

       commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
       if(!bbcNewsHindiPageObject.frontpage.isDisplayed)
       {
           System.out.println("Scrolling up")
           commonFunctionKotlin.verticalSwipe(androidDriver, "Up")
       }

       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.imageitembadge)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineauthorname)
       commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineauthortitle)
       commonFunctionKotlin.navigateBack(androidDriver)

   } catch (e:Exception ) {
   }

    @Test(priority = 3, description = "checking the india page")
    @Throws(Exception::class)
    fun testIndiaPage() {
        commonFunctionKotlin.startTest("IndiaPage", "Checking the IndiaPage", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_india, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_india.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.imageitembadge)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineauthorname)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineauthortitle)

        commonFunctionKotlin.scrolltoElement(androidDriver, bbcNewsHindiPageObject.relatedtopics)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.relatedarticles)

        commonFunctionKotlin.navigateBack(androidDriver)
    }


    @Test(priority = 3, description = "checking the international page")
    @Throws(Exception::class)
    fun testInternationalPage() {
        commonFunctionKotlin.startTest("InternationalaPage", "Checking the InternationalPage", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_international, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_international.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)

        commonFunctionKotlin.navigateBack(androidDriver)
    }


    @Test(priority = 4, description = "checking the Entertainment page")
    @Throws(Exception::class)
    fun testEntertainmentPage() {
        commonFunctionKotlin.startTest("EntertainmentPage", "Checking the EntertainmentPage", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_entertainment, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_entertainment.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        //commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.imageitembadge);
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)


        commonFunctionKotlin.scrolltoElement(androidDriver, bbcNewsHindiPageObject.relatedtopics)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.relatedarticles)

        commonFunctionKotlin.navigateBack(androidDriver)
    }

    @Test(priority = 5, description = "checking the Sports page")
    @Throws(Exception::class)
    fun testSportsPage() {
        commonFunctionKotlin.startTest("Sports", "Checking the Sports", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_sports, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_sports.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.imageitembadge)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.imageitemcaption)


        commonFunctionKotlin.scrolltoElement(androidDriver, bbcNewsHindiPageObject.relatedtopics)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.relatedarticles)

        commonFunctionKotlin.navigateBack(androidDriver)
    }


    @Test(priority = 6, description = "checking the Radio page")
    @Throws(Exception::class)
    fun testRadioPage() {
        commonFunctionKotlin.startTest("Radio", "Checking the Radio", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_radio, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_radio.isSelected)

        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        Assert.assertEquals("सुनिए", bbcNewsHindiPageObject.headlinetitle.text)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)
        Assert.assertEquals("12 अप्रै 2018", bbcNewsHindiPageObject.headlineinfo.text)

        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.radiopagetext)
        Assert.assertEquals("नमस्कार भारत  (06.30IST - 07.00IST)", bbcNewsHindiPageObject.radiopagetext.text)

        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.radiopagetextdaily)
        Assert.assertEquals("दिनभर (19.30IST - 20.00IST)", bbcNewsHindiPageObject.radiopagetextdaily.text)

    }

    @Test(priority = 7, description = "checking the Science&Technology page")
    @Throws(Exception::class)
    fun testScienceTechnologyPage() {
        commonFunctionKotlin.startTest("Science&Technology", "Checking the Science&Technology", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_sciencetechnology, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_sciencetechnology.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)

        commonFunctionKotlin.scrolltoElement(androidDriver, bbcNewsHindiPageObject.relatedtopics)

        commonFunctionKotlin.navigateBack(androidDriver)

    }


    @Test(priority = 8, description = "checking the Science&Technology page")
    @Throws(Exception::class)
    fun testLookAtPage() {
        commonFunctionKotlin.startTest("LookAt", "Checking the Science&LookAt Page", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_lookat, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_lookat.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)


        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.playbutton)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mediaitemcaption)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.playbutton, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.volumebutton)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.Fullscreenbutton)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.seekbar)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.pausebutton)

        commonFunctionKotlin.navigateBack(androidDriver)

    }

    @Test(priority = 9, description = "checking the Pictures  page")
    @Throws(Exception::class)
    fun testPicturesPage() {
        commonFunctionKotlin.startTest("Pictures", "Checking the Pictures Page", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_thephotos, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_thephotos.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)


        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.picturestitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.picturessubtitle)

        for (i in 0..6) {
            val images = androidDriver.findElement(By.xpath("//android.widget.ImageView[@index='$i']"))
            images.isDisplayed
        }

        commonFunctionKotlin.navigateBack(androidDriver)

    }


    @Test(priority = 10, description = "checking the Social  page")
    @Throws(Exception::class)
    fun testSocialPage() {
        commonFunctionKotlin.startTest("Social", "Checking the Social Page", "Smoke")
        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_social, false)
        assertTrue(bbcNewsHindiPageObject.bbchindi_social.isSelected)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_name)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.mainitem_layout_last_updated)

        commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.article, false)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlinetitle)
        commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.headlineinfo)

        commonFunctionKotlin.navigateBack(androidDriver)

    }


    @Test(priority = 11, description = "Checking the MoreOptions Menu")
    @Story("MoreOptions")
    @Severity(SeverityLevel.CRITICAL)
    @Throws(Exception::class)
    fun testMenuItems() {
        try {
            commonFunctionKotlin.startTest("MoreOptions", "Checking the MoreOptions Menu", "Smoke")
            commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbc_moreoptions, false)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.bbchindi_help)
            //  commonFunctionKotlin.elementDisplayed(androidDriver,bbcNewsHindiPageObject.bbchindi_Internalsettings);
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.bbchindi_settings)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.bbchindi_pleasecontact)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.bbchindi_OtherBBCapplications)
            commonFunctionKotlin.navigateBack(androidDriver)


        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    @Test(priority = 12, description = "Checking the More Settings Options Menu")
    @Story("MoreOptions")
    @Severity(SeverityLevel.CRITICAL)
    @Throws(Exception::class)
    fun testMoreSettingsOptions() {
        try {
            commonFunctionKotlin.startTest("MoreOptions", "Checking the More Settings Options Menu", "Smoke")
            commonFunctionKotlin.tapButton(androidDriver, bbcNewsHindiPageObject.bbchindi_Moresettings, false)
            // commonFunctionKotlin.elementDisplayed(androidDriver,bbcNewsHindiPageObject.bbchindi_localnews);
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.bbchindi_topics)


        } catch (e: Exception) {
            e.printStackTrace()
        }

    }


    @Test(priority = 13, description = "Checking the More Settings Options Topics")
    @Story("MoreOptions-Topics")
    @Severity(SeverityLevel.CRITICAL)
    @Throws(Exception::class)
    fun testMoreSettingsOptions_Topics() {
        try {
            commonFunctionKotlin.startTest("MoreOptionsTopics", "Checking the More Settings Options Topics", "Smoke")
            //  commonFunctionKotlin.tapButton(androidDriver,bbcNewsHindiPageObject.bbchindi_topics_collapsegroup,false);
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindihomepage)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindibharath)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindienrairnment)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindiinternatonal)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindilookat)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindiphotos)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindiscience)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindiphotos)
            commonFunctionKotlin.elementDisplayed(androidDriver, bbcNewsHindiPageObject.hindisocial)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    @AfterMethod
    fun getResult(result: ITestResult) {
        try {
            commonFunctionKotlin.getTestResult(androidDriver, result)
        } catch (e: IOException) {
        }


    }

    @AfterTest
    fun tearDown() {
        commonFunctionKotlin.publishReport()
        androidDriver.closeApp()
        androidDriver.quit()

    }

    companion object {

        private val curDate = Date()
        private val format = SimpleDateFormat("yyyy-MM-dd")
        private val DateToStr = format.format(curDate)
        var workingDirectory = System.getProperty("user.dir")
        private val screenshotpath = workingDirectory + "/Screenshots/"
    }
}